File: mns-reactor/src/main/java/cz/fb/manaus/spring/ReactorLocalConfiguration.java
package cz.fb.manaus.spring;

import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.FilterType;
import org.springframework.stereotype.Repository;

@Configuration
@ComponentScan(value = "cz.fb.manaus.reactor",
        excludeFilters = {
                @ComponentScan.Filter(type = FilterType.ANNOTATION, value = Repository.class),
                @ComponentScan.Filter(type = FilterType.ANNOTATION, value = DatabaseComponent.class),
        }
)
public class ReactorLocalConfiguration {


}

File: mns-reactor/src/main/java/cz/fb/manaus/spring/ReactorLocalConfiguration.kt
package cz.fb.manaus.spring

import cz.fb.manaus.reactor.betting.AmountAdviser
import cz.fb.manaus.reactor.betting.FixedAmountAdviser
import cz.fb.manaus.spring.conf.BettingConf
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration

@Configuration
@ComponentScan(value = ["cz.fb.manaus.reactor"])
@EnableConfigurationProperties(BettingConf::class)
open class ReactorLocalConfiguration {

    @Bean
    open fun fixedAmountAdviser(bettingConf: BettingConf): AmountAdviser {
        val amount = bettingConf.amount
        return FixedAmountAdviser(amount)
    }
}
